### PR TITLE
426 bug msg keyerror data

### PR DESCRIPTION
**Pull request by bsuryadevara**
- Users can pass the name of a column that has to be pre-processed more easily with parameterized pre-process columns.

### CODE REVIEW

**Comment by cwharris** [comment]
rerun tests

**Review by cwharris — APPROVED** [review]

**Review by raykallen — APPROVED** [review]

File: morpheus/stages/preprocess/preprocess_nlp_stage.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ class PreprocessNLPStage(PreprocessBaseStage):
         overflowing token-ids can contain duplicated token-ids from the main sequence. If max_length is equal to stride
         there are no duplicated-id tokens. If stride is 80% of max_length, 20% of the first sequence will be repeated on
         the second sequence and so on until the entire sentence is encoded.
+    column : str
+        Name of the column containing the data that needs to be preprocessed.
```
Comment (mdemoret-nv) [inline morpheus/stages/preprocess/preprocess_nlp_stage.py:67]:
```suggestion
    column : str, default = "data"
        Name of the column containing the data that needs to be preprocessed.
```

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
We need a test case to exercise this new functionality.

File: morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp [hunk 2]
```diff
@@ -86,6 +88,7 @@ struct PreprocessNLPStageInterfaceProxy
                                                                           bool truncation,
                                                                           bool do_lower_case,
                                                                           bool add_special_token,
+                                                                          const std::string column,
```
Comment (dagardner-nv) [inline morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp:91]: I would make this a 
```suggestion
                                                                          const std::string& column,
```

**Review by dagardner-nv — APPROVED** [review]
LGTM 

minor nitpick to make one of the arguments a reference, but looks good in general.

File: morpheus/stages/preprocess/preprocess_nlp_stage.py (hunk 1 above)
Comment (bsuryadevara) [reply morpheus/stages/preprocess/preprocess_nlp_stage.py:67]: @mdemoret-nv I have updated the existing tests by adding a column argument to PreprocessNLPStage instantiation. Please let me know if I need to create a new function.

File: morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp (hunk 2 above)
Comment (mdemoret-nv) [reply morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp:91]: This is more than a nitpick. Good catch. We should either be using pass-by-value and move semantics, or const-ref.

File: morpheus/stages/preprocess/preprocess_nlp_stage.py (hunk 1 above)
Comment (mdemoret-nv) [reply morpheus/stages/preprocess/preprocess_nlp_stage.py:67]: This is more what I had in mind: https://github.com/nv-morpheus/Morpheus/pull/428/files#diff-b044cf5578e03f4186927f46fd55f286a8eea240249154287735b22c989f6b43R76

File: morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp [hunk 3]
```diff
@@ -54,6 +54,7 @@ class PreprocessNLPStage
                        bool truncation,
                        bool do_lower_case,
                        bool add_special_token,
+                       std::string column,
```
Comment (mdemoret-nv) [inline morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp:57]: Why does this come before `stride`? If it's due to the default value, we can use `"data"` for the default. If we change the order of the arguments, we need to mark this as breaking.

Can you move the `column` option to be after `stride` everywhere?

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Need to have the `column` argument come after `stride` otherwise this is a breaking change.

File: morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp (hunk 3 above)
Comment (bsuryadevara) [reply morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp:57]: sure

Comment (bsuryadevara) [reply morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp]: @mdemoret-nv changed the order as suggested

Comment (mdemoret-nv) [reply morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp]: Great thanks.

**Comment by mdemoret-nv** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
\rerun tests

**Review by mdemoret-nv — APPROVED** [review]
New tests look good. CI has been verified to complete locally without any issues.

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**